bokehjs/src/lib/models/glyphs/glyph.ts

```diff
@@ -368,7 +368,17 @@ export abstract class GlyphView extends DOMComponentView {
     }
   }
 
-  protected _transform_array<T>(prop: p.BaseCoordinateSpec<T>, array: Arrayable<unknown>) {
+  protected transform_array<T>(prop: p.BaseCoordinateSpec<T>, array: Arrayable<unknown>) {
+    // examine just the top level of a 2-d array to validate
+    // that every subitem is an array of some kind, as expected
+    if (prop instanceof p.CoordinateSeqSpec) {
+      for (let i = 0; i < array.length; i++) {
+        if (!isArrayable(array[i])) {
+          logger.error(`expected a 2-d array for ${this.model.type}.${prop.attr}`)
```

Comment:
Note that `logger.error()` doesn't stop execution like throwing an exception does, so this may log a lot if every index is bad and then it will fail anyway at some point.

Reply:
My previous suggestion for using `logger.error()` was to combine it with error recovery. If no recovery is performed, then I would throw an exception as originally proposed.